src/videodownlinkbandwidthpolicy/VideoAdaptiveProbePolicy.ts

```diff
@@ -49,14 +49,4 @@ export default class VideoAdaptiveProbePolicy extends VideoPriorityBasedPolicy {
       this.videoPreferences = undefined;
     }
   }
-
-  /**
-   * [[VideoAdaptiveProbePolicy]] does not allow setting video preferences and this function
-   * will be a no-op.  Please use [[VideoPriorityBasedPolicy]] directly if you would like to set
-   * preferences.
-   */
-  chooseRemoteVideoSources(_preferences: VideoPreferences): void {
```

Comment:
Adaptive auto manage the preference so this is a change in behavior. Should we just truncate here instead?

Reply:
I accidentally deleted that function in VideoAdaptivePolicy. This class needs to stay, else it will be similar to PriorityBasedDL.

And yes, we should just truncate here.

Reply:
Also updateIndex already update the preference (which one happens first? updateIndex or remoteVideoSourcesDidChange do you know ?) so there is a conflict here.

Reply:
1. The UpdateIndex happens first and it provides all the new index values, which will be used by the `chooseSubscription` function.
2. The `chooseSubscription` function is called in the `ReceiveVideoStreamIndexTask` in the `resubscribe` function
3. In the `resubscribe` function we truncate the list to have only 25 videos in the `videoToReceive` list
4. Then finally the remoteVideoSourcesDidChange function is called incase, anything has changed.